### ISSUE TITLE
Navigating to documents hidden in the node-tree breaks the link to the current page

### ISSUE THREAD

**Issue by PRGfx** [neos/neos-ui#2954]
### Description

When navigating to a document hidden in the node-tree (i.e. through links in the guest-frame), the link to the previously selected document in the node-tree does no longer work.

### Steps to Reproduce

1. Setup neos-base-distribution
2. Site-Package and site: https://gist.github.com/PRGfx/a6f33ddc2a53bd1e8b91f821aed6b6dc
    ```
    home (Broken.NodeTree:Document.Home)
    + directory (Broken.NodeTree:Document.Page)
    | + special 1 (Broken.NodeTree:Document.Special)
    | + special 2 (Broken.NodeTree:Document.Special)
    + directory 2 (Broken.NodeTree:Document.Page)
    ```
    - `Page` lists all pages of type `Special` on the site and links to them
    - rendering only extends `Neos.Neos:Page` and only outputs the document title
    - the node-tree is set to render `Neos.Neos:Document,!Broken.NodeTree:Document.Special`
3. Navigate
    - `directory` > `special 1` > `directory` fails
    - `directory` > `special 1` > `directory 2` works
    - `directory 2` > `special 1` > `directory 2` fails

#### Expected behavior

The directory page should load again

#### Actual behavior

- The page does not load, a `/neos/preview?node[__contextNodePath]=…` request is not fired
- the inspector properties are loaded for the correct page

Navigating back to the parent page fails

https://user-images.githubusercontent.com/3868790/133447742-e4337fe9-4873-4b0f-9353-93fc58e250fe.mp4

Navigating back to a different page works as expected

https://user-images.githubusercontent.com/3868790/133447913-e6ac5fe9-5dbc-49c1-91ef-4ee8a2220519.mp4

Navigating back to the previous page fails; the parent/sub-page relation does not matter

https://user-images.githubusercontent.com/3868790/133447936-9dd2e457-eac9-450f-906f-f42bce805032.mp4

### Affected Versions
tested with

Neos: 7.1
UI: 7.1
